src/server/Agent.js

```diff
@@ -72,36 +82,23 @@ export default class Agent extends SideAgent {
     // Take care of the given accepted connection.
     // The connection may have been accepted by another Agent instance.
     _startServing(userSocket) {
-        let transaction = null;
-        if (this._transaction.started()) {
-            transaction = new Transaction(this);
-
-            // reset() copies everything, so this cannot work well if
-            // the original response has transaction-specific pieces
-            assert(!this._originalResponse.finalized());
-            transaction.response.reset(this._originalResponse);
-        } else {
-            transaction = this._transaction;
-
-            // see the reset(this._originalResponse) comments above
-            assert(!transaction.response.finalized());
-            assert(!this._originalResponse);
-            this._originalResponse = transaction.response.clone();
-        }
-
-        return this._runTransaction(transaction, userSocket);
+        assert(!this._transaction.started());
+        assert(!this._transaction.response.finalized());
```

Comment:
I examined the code once more and concluded that we do not need this assertion because there should be no situation when we re-use a response which once was finalized. In server.Agent we always have a brand-new transaction object (and only one instance of) and, obviously, its response cannot be finalized before we start that transaction (the finalization happens in transaction.makeMessage() after calling transaction.send()).

Reply:
But what if somebody calls server.response().finalize() before starting the server? Would that break something? Without hitting other/meaningful assertions?

Reply:
I performed such a test and got an assertion in `Transaction.fillMessageBody()`. It looks like that normally Transaction.makeMessage() never goes to the last condition, because Transaction.fillMessageBody() always asserts (in is not re-defined in server.Transaction or client.Transaction).

Reply:
Yes, fillMessageBody() looks like a leftover from the original design where we expected test cases to sublass Transaction (instead of configuring it). We should not polish that in this PR.

After looking through sources, I am content with going without this assertion. The assertion is correct -- transaction code does not handle pre-finalized messages correctly. However, this code is no longer the right place for this assertion, and more changes are needed to fix transaction code (e.g., we call this.request.header.addByDefault() after finalize() which  leaves the added header field (if any) unfinalized). Those changes are probably complex and are outside this PR scope; I am pretty sure we should get rid of finalization (as a single step in message lifetime).